modules/renter/consts.go

```diff
@@ -111,6 +111,14 @@ var (
 		Testing:  3 * time.Second,
 	}).(time.Duration)
 
+	// remoteRepairDownloadThreshold defines the threshold in percent under
+	// which the renter starts repairing a file that is not available on disk.
+	remoteRepairDownloadThreshold = build.Select(build.Var{
+		Dev:      0.0,  // always download
+		Standard: 0.25, // download if at least 25% are missing
+		Testing:  0.0,  // always download
```

Comment:
if testing and dev are always `0`, you have no coverage of the case where where it's not supposed to download a file to do a repair. I think my preference is for `0.25`, because it limits the number of hosts that you need in order to have a file that's missing less than 25% of its redundancy but more than 0% of its redundancy.

Reply:
@DavidVorick the problem is that introduces non deterministic behavior when waiting for the redundancy to increment. If we want to repair 5 chunks by downloading them first and not all of them are added to the heap in one iteration of the repair loop, the chunk that ends up in the next iteration might not be downloaded because the redundancy is already too high.
The only other workaround would be to export this field and only check for 75% redundancy in the test.

Reply:
Seems reasonable to me to export the field and allow the test package to only check it goes above that redundancy.

Are you sure that is the problem though? If it gets started on a chunk it should take it to 100%. If it doesn't get started on a chunk, the chunk shouldn't have any redundancy missing on the next iteration.

To the best of my knowledge, it shouldn't drop a chunk that's halfway complete, once you pass pieces out to workers they don't get interrupted unless they are determined to be bad workers.

Reply:
Yes I'm quite sure. It usually happened in the following scenario:
The group has 4 offline hosts and 5 online hosts. The renter has seen all of those hosts and has contracts with all of them but the worker pool was not updated yet and contains only 8 hosts.
The first iteration of the repair starts but can not upload the last piece since one of the hosts is still missing from the pool.
Before the second iteration starts we update the pool and it now contains 9 workers. The chunks are queued again but this time they have too many completed pieces to be downloaded again.